### PR TITLE
feat!: default to semver-coerced instead of semver

### PR DESCRIPTION
**Pull request by secustor**
## Changes
Use `semver-coerced` as default versioning instead of the strict `semver`. 

## Context
https://renovatebot.slack.com/archives/CAFH752JU/p1676991210999899

I still have to identify places where we are referring to `semver` as default versioning. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — APPROVED** [review]

**Comment by secustor** [comment]
@HonkingGoose Do you know maybe of other sections which we have to change? I have done so far a simple search for "semver".

**Comment by rarkins** [comment]
V35 maybe?

File: lib/modules/datasource/gitlab-tags/readme.md [hunk 1]
```diff
@@ -32,4 +32,4 @@ Now you may use comments in your `versions.ini` files to automatically update de
 NKJS_VERSION=3.4.0
```
Comment (HonkingGoose) [inline lib/modules/datasource/gitlab-tags/readme.md:32]: We have some docs about semver-coerced in `lib/modules/versioning/semver-coerced/readme.md`. That file needs some changes:

- Follow one sentence per line pattern
- Check if the information is correct now that you've changed the default behavior

File: lib/modules/datasource/repology/readme.md [hunk 2]
```diff
@@ -50,4 +50,4 @@ When the operating system package for `gcc` of `Alpine Linux 3.12` is updated, R
 <!-- prettier-ignore -->
```
Comment (HonkingGoose) [inline lib/modules/datasource/repology/readme.md:50]: I'd also search for just the word `semantic` to find other mentions of the concept.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/datasource/gitlab-tags/readme.md (hunk 1 above)
Comment (secustor) [reply lib/modules/datasource/gitlab-tags/readme.md:32]: Adapted following your input. 
One note as I'm not sure if this is clear. I haven't changed the default **of** `semver-coerced` but this changes the default versioning from `semver` to `semver-coerced`.

File: lib/modules/datasource/repology/readme.md (hunk 2 above)
Comment (secustor) [reply lib/modules/datasource/repology/readme.md:50]: I have searched for semantic, but couldn't find anything relevant.

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**